internal/fileutil/fileutil.go

```diff
@@ -48,3 +50,34 @@ func AtomicWriteFile(filename string, reader io.Reader, mode os.FileMode) error
 
 	return fs.RenameWithFallback(tempName, filename)
 }
+
+// LockedAtomicWriteFile atomically writes a file to disk with file locking to prevent
+// concurrent writes. This is particularly useful on Windows where concurrent writes
+// to the same file can cause "Access Denied" errors.
+//
+// The function acquires a lock on the target file, checks if it already exists,
+// and only writes if it doesn't exist. This prevents duplicate work when multiple
+// processes try to write the same file simultaneously.
+//
+// Returns true if the file was written, false if it already existed.
+func LockedAtomicWriteFile(filename string, reader io.Reader, mode os.FileMode) (bool, error) {
+	// Acquire a file lock for this specific file path to prevent concurrent writes
+	fileLock := flock.New(filename + ".lock")
+	if err := fileLock.Lock(); err != nil {
+		return false, err
+	}
+	defer fileLock.Unlock()
+
+	// Check if the file already exists (another process might have already written it)
```

Comment:
If you use TryLock, you'd probably want to change this to look the returned `FileInfo.Size()` here.